db/row_cache.cc

```diff
@@ -1251,6 +1252,10 @@ future<> row_cache::invalidate(external_updater eu, dht::partition_range_vector&
                                     });
                                     break;
                                 }
+                              } else {
+                                _tracker.clear_continuity(*it);
+                                ++it;
```

Comment:
we can have a large run of keys that don't filter in, e.g. when migrating a tablet in. couldn't this lead to stalls since we don't preempt in this path? shouldn't we do the preemption check in both cases?

Reply:
The fix is pretty easy, I think we do not need to wait. I opened a PR: https://github.com/scylladb/scylladb/pull/26053. Please review and consider merging.